pulp_rpm/app/migrations/0004_jsonfields.py

```diff
@@ -0,0 +1,301 @@
+# Generated by Django 2.2.5 on 2019-09-26 20:53
+"""
+This migration will follow these steps:
+1 - create a new json field
+2 - create a data migration
+3 - remove old text field
+4 - rename the json field
+"""
+import json
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations
+
+
+def forwards_func(apps, schema):
+    Category = apps.get_model('rpm', 'Category')
+    for category in Category.objects.all():
+        try:
+            category.json_group_ids = json.loads(category.group_ids)
+            category.save()
+        except json.decoder.JSONDecodeError:
+            print('Cannot convert {} object'.format(category.pk))
+
+    Environment = apps.get_model('rpm', 'Environment')
+    for environment in Environment.objects.all():
+        try:
+            environment.json_group_ids = json.loads(environment.group_ids)
+            environment.json_option_ids = json.loads(environment.option_ids)
+            environment.save()
+        except json.decoder.JSONDecodeError:
+            print('Cannot convert {} object'.format(environment.pk))
+
+    Modulemd = apps.get_model('rpm', 'Modulemd')
+    for modulemd in Modulemd.objects.all():
+        try:
+            modulemd.json_artifacts = json.loads(modulemd.artifacts)
+            modulemd.json_dependencies = json.loads(modulemd.dependencies)
+            modulemd.save()
+        except json.decoder.JSONDecodeError:
+            print('Cannot convert {} object'.format(modulemd.pk))
+
+    ModulemdDefaults = apps.get_model('rpm', 'ModulemdDefaults')
+    for modulemd_defaults in ModulemdDefaults.objects.all():
+        try:
+            modulemd_defaults.json_profiles = json.loads(modulemd_defaults.profiles)
+            modulemd_defaults.save()
+        except json.decoder.JSONDecodeError:
+            print('Cannot convert {} object'.format(modulemd_defaults.pk))
+
+    Package = apps.get_model('rpm', 'Package')
+    for package in Package.objects.all():
+        try:
+            package.json_requires = json.loads(package.requires)
+            package.json_provides = json.loads(package.provides)
+            package.json_conflicts = json.loads(package.conflicts)
+            package.json_obsoletes = json.loads(package.obsoletes)
+            package.json_suggests = json.loads(package.suggests)
+            package.json_enhances = json.loads(package.enhances)
+            package.json_recommends = json.loads(package.recommends)
+            package.json_supplements = json.loads(package.supplements)
+            package.save()
+        except json.decoder.JSONDecodeError:
```

Comment:
What happens with the Package field record in case of JSONDecodeError? Will it be empty because the text field is removed and the json one is added?

I'm not sure why do we need a data migration in the beta release. We risk introducing bugs in it if we claim that we support an upgrade path to the next beta, and it's very likely we'll need to regenerate migrations.
Can anyone clarify? I probably miss something. @fabricio-aguiar  @dralley

Reply:
good points @goosemania doing all-in-one migration like that probably would result in leaving the field empty.
About the data migration, I was thinking about users who already have some data on these textfields, but I forgot it was a beta release.